[Index Management] Refactor IndexActionsContextMenu

damian-polewski opened the pull request:
## Summary

Closes #236071.

Refactors `IndexActionsContextMenu` to TypeScript and changes it from class component to functional component.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

damian-polewski committed:
feat: refactor index_actions_context_menu

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343675)
* Commit: f767a6d7c2803fd680dcace81ba99c28fbbd0ad9

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/343675#0199808d-9ac9-40ca-97a5-c60d8c06f7d6)

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343675","number":343675,"commit":"f767a6d7c2803fd680dcace81ba99c28fbbd0ad9"}],"number":343675}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into index_management/refactor_index_actions_context_menu

damian-polewski committed:
fix: ts error

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343700)
* Commit: 1b5aa6797f17646816b3e47767084958685c506e

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/343700#019980ac-ea88-4043-9cb7-f02cb69cfcdf)

### History
* :broken_heart: [Build #343675](https://buildkite.com/elastic/kibana-pull-request/builds/343675) failed f767a6d7c2803fd680dcace81ba99c28fbbd0ad9

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343700","number":343700,"commit":"1b5aa6797f17646816b3e47767084958685c506e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343675","number":343675,"commit":"f767a6d7c2803fd680dcace81ba99c28fbbd0ad9"}],"number":343700}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343912)
* Commit: 014fc9bf0d62c25e74af73af1ebdc1fd8c43592b

### Failed CI Steps
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad03-45f0-ae7c-80b30daa4d05)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981f3-6173-4101-9287-434a6d7044d6)
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad17-477e-bcbd-f35e1498c277)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad03-45f0-ae7c-80b30daa4d05) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981cf-ad03-45f0-ae7c-80b30daa4d05/artifacts/019981f3-3ef5-4d78-a3b7-328f6e12a97b) Jest Tests #<span></span>15 / <IndexDetailsPage /> context menu forcemerges an index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981f3-6173-4101-9287-434a6d7044d6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981f3-6173-4101-9287-434a6d7044d6/artifacts/019981fe-c2bc-4a17-a3b1-329ebd6df6f1) Jest Tests #<span></span>15 / <IndexDetailsPage /> context menu forcemerges an index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad03-45f0-ae7c-80b30daa4d05) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981cf-ad03-45f0-ae7c-80b30daa4d05/artifacts/019981f3-3ef7-47c7-853b-dc577d05330e) Jest Tests #<span></span>15 / <IndexManagementHome /> index actions should be able to force merge an index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981f3-6173-4101-9287-434a6d7044d6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981f3-6173-4101-9287-434a6d7044d6/artifacts/019981fe-c2bc-49cf-a975-7ea45d0c85ce) Jest Tests #<span></span>15 / <IndexManagementHome /> index actions should be able to force merge an index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad03-45f0-ae7c-80b30daa4d05) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981cf-ad03-45f0-ae7c-80b30daa4d05/artifacts/019981f3-3ef6-4699-8d4a-a88be736ede2) Jest Tests #<span></span>15 / index table force merge button works from context menu
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981f3-6173-4101-9287-434a6d7044d6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981f3-6173-4101-9287-434a6d7044d6/artifacts/019981fe-c2bd-440f-9efb-64f477aaeb14) Jest Tests #<span></span>15 / index table force merge button works from context menu
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343912#019981cf-ad17-477e-bcbd-f35e1498c277) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343912/jobs/019981cf-ad17-477e-bcbd-f35e1498c277/artifacts/019981f8-b731-4e77-8bbc-1073d6b23721) FTR Configs #<span></span>14 / Space awareness actions GET /agents/action_status should return agent actions in a custom space

### History
* :broken_heart: [Build #343700](https://buildkite.com/elastic/kibana-pull-request/builds/343700) failed 1b5aa6797f17646816b3e47767084958685c506e
* :broken_heart: [Build #343675](https://buildkite.com/elastic/kibana-pull-request/builds/343675) failed f767a6d7c2803fd680dcace81ba99c28fbbd0ad9

cc @damian-polewski

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343912","number":343912,"commit":"014fc9bf0d62c25e74af73af1ebdc1fd8c43592b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343700","number":343700,"commit":"1b5aa6797f17646816b3e47767084958685c506e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343675","number":343675,"commit":"f767a6d7c2803fd680dcace81ba99c28fbbd0ad9"}],"number":343912}
buildkite-pr-comment-->

damian-polewski committed:
Merge branch 'main' into index_management/refactor_index_actions_context_menu

kapral18 committed:
chore: address comments
- fix the bug with forcemerge segments modal using stale state
- future proof other modals by not storing their renderers in state
but storing the keys representing renderers instead
- extract ModalHost component out of modal
- refactor IndexActionsContextMenu to use ModalHost
- add tests for IndexActionsContextMenu and ModalHost

kapral18 committed:
refactor: update index actions context menu and modal host tests

- Remove explicit mock for `modal_host` as it's brittle
- Update force merge modal logic to default to '1' segment when input is empty, to make integration tests pass
- Improve test titles structure for clarity.

kapral18 committed:
Merge branch 'main' into index_management/refactor_index_actions_context_menu

damian-polewski committed:
fix: use user-event

sabarasaba reviewed (approved):
skimmed through all the changes, the changes made lgtm

damian-polewski committed:
fix: close popover when navigating to a different page

kapral18 reviewed (approved):
Great job. Thanks!
